### ISSUE TITLE
Silent data loss: Last line is ignored if it does not end with newline

### ISSUE THREAD

**Issue by vog** [parsecsv/parsecsv-for-php#85]
The CSV parser causes a silent loss of data if an input CSV file does not have a trailing newline. In that case, the last line is ignored without any warning.

Example program:

``` php
<?php
$test_csv = "a1;b1;c1;d1\na2;b2;c2;d2";
require 'parsecsv.lib.php';
$parser = new parseCSV();
$parser->encoding('UTF-8', 'UTF-8');
$parser->heading = false;
$parser->delimiter = ';';
$parser->parse($test_csv);
var_export($parser->data);
```

Expected result:

``` php
array (
  0 => 
  array (
    0 => 'a1',
    1 => 'b1',
    2 => 'c1',
    3 => 'd1',
  ),
  1 => 
  array (
    0 => 'a2',
    1 => 'b2',
    2 => 'c2',
    3 => 'd2',
  ),
)
```

Actual result:

``` php
array (
  0 => 
  array (
    0 => 'a1',
    1 => 'b1',
    2 => 'c1',
    3 => 'd1',
  ),
)
```

**Comment by Piskvor** [comment]
Cannot reproduce with current version, I get the expected result.

**Comment by vog** [comment]
Indeed, it seems this issue was fixed in commit cff692d97665209e6440f0913645babd4c0268d4
